### PR TITLE
add prometheus operator as a part of CI

### PR DESCRIPTION
**Pull request by SamYuan1990**
try to add prometheus operator as a part of CI

reference
https://prometheus-operator.dev/docs/prologue/quick-start/#deploy-kube-prometheus

if we just deploy with bundle(https://github.com/prometheus-operator/prometheus-operator), it will not create namespace as monitor.
hence, we deploy quick start as an all in one solution in CI.

Signed-off-by: Sam Yuan <yy19902439@126.com>

### CODE REVIEW

**Comment by SamYuan1990** [comment]
it's a draft so far. will covert to ready status once code complete.

**Comment by SamYuan1990** [comment]
resolve #181

**Comment by SamYuan1990** [comment]
@rootfs , please help re run CI check. fails at 
```
Step 14/21 : FROM quay.io/sustainable_computing_io/kepler_base:latest
Head "https://quay.io/v2/sustainable_computing_io/kepler_base/manifests/latest": Get "https://quay.io/v2/auth?scope=repository%3Asustainable_computing_io%2Fkepler_base%3Apull&service=quay.io": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
make[1]: *** [Makefile:84: _build_containerized] Error 1
make[1]: Leaving directory '/home/runner/work/kepler/kepler'
make: *** [Makefile:165: cluster-sync] Error 2
```

**Review by rootfs — APPROVED** [review]

File: .github/workflows/integration_test.yml [hunk 1]
```diff
@@ -22,6 +22,14 @@ jobs:
         run: make cluster-up
         env:
             CLUSTER_PROVIDER: ${{matrix.kube_provider}}
+      - name: Checkout kube-operator repo
+        uses: actions/checkout@v3
+        with:
+          repository: prometheus-operator/kube-prometheus
+          path: kube-prometheus
+      - name: deploy kube operator
+        run:  ls && kubectl create -f manifests/setup && until kubectl get servicemonitors --all-namespaces ; do date; sleep 1; echo ""; done && kubectl create -f manifests/
```
Comment (marceloamaral) [inline .github/workflows/integration_test.yml:31]: we should avoid to have a big logics in the here

Instead, we should create a script in the `automation` folder to run this logic in the CI

**Pull request merged**